consent-ws/src/main/java/org/broadinstitute/consent/http/service/DatabaseDataAccessRequestAPI.java

```diff
@@ -603,4 +605,30 @@ private Document processDataSet(Document dataAccessRequest, ConsentDataSet conse
         return dataAccess;
     }
 
-}
+    private void updateResearcherIdentification(Document dataAccessRequest) {
+        Integer userId = dataAccessRequest.getInteger(DarConstants.USER_ID);
+        String linkedIn = dataAccessRequest.getString(ResearcherFields.LINKEDIN_PROFILE.getValue());
+        String orcId = dataAccessRequest.getString(ResearcherFields.ORCID.getValue());
+        String researcherGate = dataAccessRequest.getString(ResearcherFields.RESEARCHER_GATE.getValue());
+
+        if (!Objects.equals(linkedIn, researcherPropertyDAO.findPropertyValueByPK(userId, ResearcherFields.LINKEDIN_PROFILE.getValue()))) {
+            researcherPropertyDAO.deletePropertyByUser(ResearcherFields.LINKEDIN_PROFILE.getValue(), userId);
+            if (!StringUtils.isEmpty(linkedIn)) {
+                researcherPropertyDAO.insertPropertyByUser(ResearcherFields.LINKEDIN_PROFILE.getValue(), linkedIn, userId);
+            }
+        }
+        if (!Objects.equals(orcId, researcherPropertyDAO.findPropertyValueByPK(userId, ResearcherFields.ORCID.getValue()))) {
+            researcherPropertyDAO.deletePropertyByUser(ResearcherFields.ORCID.getValue(), userId);
+            if (!StringUtils.isEmpty(orcId)) {
+                researcherPropertyDAO.insertPropertyByUser(ResearcherFields.ORCID.getValue(), orcId, userId);
+            }
+        }
+        if (!Objects.equals(researcherGate, researcherPropertyDAO.findPropertyValueByPK(userId, ResearcherFields.RESEARCHER_GATE.getValue()))) {
+            researcherPropertyDAO.deletePropertyByUser(ResearcherFields.RESEARCHER_GATE.getValue(), userId);
+            if (!StringUtils.isEmpty(researcherGate)) {
+                researcherPropertyDAO.insertPropertyByUser(ResearcherFields.RESEARCHER_GATE.getValue(), researcherGate, userId);
+            }
+        }
+    }
```

Comment:
I think this could be simpler with a straight delete of the 3 values for the user (in one delete query) and then re-adding them (if they exist in the new data access request.) This would be a slight refactor. In this case, you're running minimum 3, potentially 3 more queries. My suggestion would be minimum 1, potentially 3 more queries.